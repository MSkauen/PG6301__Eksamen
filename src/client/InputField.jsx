import React from "react";

export function InputField({ label, onChangeValue, value, type }) {
  return (
    <div>
      <label>
        {label}:{" "}
        <input
          type={type}
          value={value}
          onChange={(e) => onChangeValue(e.target.value)}
        />
      </label>
    </div>
  );
}
