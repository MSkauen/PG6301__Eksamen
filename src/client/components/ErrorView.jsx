import React from "react";

export function ErrorView({ error, reload }) {
  if (error.status === 401) {
    return (
      <div>
        You are not logged in.
        <a href={"/api/login"} target={"_self"}>
          <button>Log in</button>
        </a>
      </div>
    );
  }
  return (
    <>
      <div>Something went wrong: {error.toString()}</div>
      {reload && <button onClick={reload}>Try Again</button>}
    </>
  );
}
