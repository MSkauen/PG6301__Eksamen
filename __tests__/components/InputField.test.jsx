import { InputField } from "../../src/client/components/InputField";
import TestRenderer from "react-test-renderer";

import React from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
//export function InputField({ label, onChangeValue, value, type }) {
describe("inputfield", () => {
  it("show input field label on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<InputField label={"test"} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("label").textContent).toEqual("test: ");
  });

  it("show input field onChangeValue on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<InputField value={"inputvalue"} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("input").value).toEqual("inputvalue");
  });

  it("show input field type on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<InputField type={"date"} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("div").innerHTML).toContain("date");
  });
});
