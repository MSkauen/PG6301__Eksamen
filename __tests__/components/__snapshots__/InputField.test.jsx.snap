// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputfield show input field label on dom 1`] = `"<div><label>test: <input value=\\"\\"></label></div>"`;

exports[`inputfield show input field onChangeValue on dom 1`] = `"<div><label>: <input value=\\"inputvalue\\"></label></div>"`;

exports[`inputfield show input field type on dom 1`] = `"<div><label>: <input type=\\"date\\" value=\\"\\"></label></div>"`;
